src/Infrastructure/Injector/SimpleInjector.php

```diff
@@ -325,7 +325,9 @@ private function resolve_argument(
 
 			$type = $parameter->getType();
 
-			if ( null === $type || $type->isBuiltin() ) {
+			if ( null === $type ||
+				( is_a( $type, 'ReflectionType' ) && method_exists( $type, 'isBuiltin' ) && $type->isBuiltin() )
+			) {
```

Comment:
I see that this change was also necessary due to phpstan on PHP 8 failing with:

```
 ------ ----------------------------------------------------------
  Line   src/Infrastructure/Injector/SimpleInjector.php
 ------ ----------------------------------------------------------
  328    Call to an undefined method ReflectionType::isBuiltin().
 ------ ----------------------------------------------------------
```

According to the PHP [docs](https://www.php.net/manual/en/class.reflectiontype.php) the `isBuiltin()` method has been moved to the `ReflectionNamedType` class in PHP 8:

> ReflectionType has become abstract and ReflectionType::isBuiltin() has been moved to [ReflectionNamedType::isBuiltin()](https://www.php.net/manual/en/reflectionnamedtype.isbuiltin.php).

Reply:
I've opened https://github.com/ampproject/amp-wap/pull/6466 to bump the PHP version to 8.0, amongst other related changes.